src/promptflow/promptflow/_sdk/_configuration.py

```diff
@@ -0,0 +1,63 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import os.path
+import uuid
+from pathlib import Path
+from typing import Optional
+
+import pydash
+
+from promptflow._sdk._utils import dump_yaml
+
+
+class Configuration(object):
+
+    CONFIG_PATH = Path.home() / ".promptflow" / "pf.yaml"
+    COLLECT_TELEMETRY = "cli.collect_telemetry"
+    USER_ID = "cli.user_session_id"
+
+    def __init__(self):
+        self.config = {}
+        if not os.path.exists(self.CONFIG_PATH.parent):
+            os.makedirs(self.CONFIG_PATH.parent, exist_ok=True)
+        if not os.path.exists(self.CONFIG_PATH):
+            with open(self.CONFIG_PATH, "w") as f:
+                f.write(dump_yaml(self.config))
+
+    @classmethod
+    def get_instance(cls):
+        return Configuration()
+
+    def _set_config(self, key, value):
+        """Store config in file to avoid concurrent write."""
+        pydash.set_(self.config, key, value)
+        with open(self.CONFIG_PATH, "w") as f:
```

Comment:
We should be careful, there might be concurrent writters.

Reply:
Offline synced. Updated `get_instance` to make sure there will be only 1 config instance for each process. And multi processing wirte is a cornet case. We'll leave as it is for now.